pyensembl/database.py

```diff
@@ -203,30 +233,30 @@ def _create_database(self, force=False):
 
     def _connect_if_exists(self):
         """
-        Return the connection if the DB exists, and otherwise return
-        None. As a side effect, stores the database connection in
-        self._connection.
+        Return the connection if the database collection exists, and
+        otherwise return None. As a side effect, stores the database 
+        connection in self._connection.
         """
         if self._connection is None:
-            db_path = self.local_db_path()
-            if exists(db_path):
-                # since version metadata is filled in last in datacache,
-                # checking for a version also implicitly checks for
-                # having finishing fill in all the tables/rows.
-                #
-                # TODO: expose this more explicitly in datacache
-                #
-                self._connection = datacache.connect_if_correct_version(
-                    db_path, DATABASE_SCHEMA_VERSION)
+            # since version metadata is filled in last in datacache,
+            # checking for a version also implicitly checks for
+            # having finishing fill in all the tables/rows.
+            #
+            # TODO: expose this more explicitly in datacache
+            #
+            self._connection = datacache.connect_if_correct_version(
+                db_url=self.db_url,
+                collection_name=self.collection_name,
+                subdir=CACHE_SUBDIR,
+                version=DATABASE_SCHEMA_VERSION)
         return self._connection
 
     @property
     def connection(self):
         """
-        Return the sqlite3 database for this Ensembl release
-        (download and/or construct it if necessary, if auto_download
-        is on). As a side effect, stores the database connection in
-        self._connection.
+        Return the database for this Ensembl release (download and/or 
+        construct it if necessary, if auto_download is on). As a side 
+        effect, stores the database connection in self._connection.
```

Comment:
I think adding explicitly the return type to the docstring (is it a sql alchemy connection, or something else?) would make this clearer